src/dune_rules/modules_field_evaluator.ml

```diff
@@ -229,10 +229,9 @@ let check_invalid_module_listing ~stanza_loc ~modules_without_implementation
       [ Pp.text "You must provide an implementation for all of these modules." ];
     (* Checking that (modules) includes all declared modules *)
     let print_undelared_modules field mods =
-      (* TODO: this is a warning for now, change to an error in 3.9. *)
       (* If we are in a vendored stanza we do nothing. *)
       if not is_vendored then
-        print ~is_error:false
+        print
```

Comment:
NB this should be ~is_error:(lang >= 3.9)

Reply:
I don't have access to the version so easily. What exactly is the point of version checking here? We did a warning stage for a version and now we wish to make it an error. What do we gain by allowing for the wrong configuration?

Reply:
It would break some projects, in the sense that upgrading the dune package would suddenly make them not build anymore.